util/packaging/common/build_helpers.sh

```diff
@@ -95,6 +109,14 @@ __build_image() {
 
   pushd ${package_dir}
 
+  # Try to pull the latest version of this image, as a best effort.
+  docker pull $docker_image_base || true
+  # Whether we pulled successfully or not, use the image SHA256 digest to
+  # identify the copy of it we have locally, so build can proceed with an old
+  # version.
```

Comment:
I think you are still working on this, just commenting so it doesn't get forgotten, but __build_image and __build_packages have diverged with respect to `docker pull`

Reply:
Yeah I really confused myself with git so switched it around a lot. I don't see `__build_image` being used anywhere, should I still apply the changes from `__build_packages` to it as well?

Reply:
Yes. `__build_image` doesn't get used by nightly, its a debugging/development tool. So arguably it doesn't need the tag stuff at all, but I think the symmetry would make things clearer.

Reply:
Makes sense. To make things symmetrical we'd need an image+tag name input to `__build_image`, is it alright if I go ahead and just add such an argument and expect it to be passed in by users?

Reply:
Thats just requiring `__build_image` to take an arg like `ubuntu:22` right? Makes perfect sense to me

Reply:
Yep 👍 

What I've been calling the image "base name" to differentiate it from the name qualified with a SHA, which isn't great terminology.